src/System.Xml.XmlSerializer/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -0,0 +1,1242 @@
+                        }
+                        ReadEndElement();
+                    }
+
+                    SetCollectionObjectWithCollectionMember(ref o, collectionMember, collectionType);
+
+                }
+            }
+        }
+
+        private object WritePrimitive(TypeMapping mapping, Func<string> readFunc)
+        {
+            if (mapping is EnumMapping)
+            {
+                if (mapping.IsSoap)
+                {
+                    throw new PlatformNotSupportedException("mapping.IsSoap");
+                }
+
+                return WriteEnumMethod((EnumMapping)mapping, readFunc);
+            }
+            else if (mapping.TypeDesc == StringTypeDesc)
+            {
+                return readFunc();
+            }
+            else if (mapping.TypeDesc.FormatterName == "String")
+            {
+                if (mapping.TypeDesc.CollapseWhitespace)
+                {
+                    return CollapseWhitespace(readFunc());
+                }
+                else
+                {
+                    return readFunc();
+                }
+            }
+            else
+            {
+                string methodName = $"To{mapping.TypeDesc.FormatterName}";
+                if (!mapping.TypeDesc.HasCustomFormatter)
+                {
+                    MethodInfo method = typeof(XmlConvert).GetMethod(methodName, BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, new Type[] { typeof(string) });
+                    return method.Invoke(null, new object[] { readFunc() });
+                }
+                else
+                {
+                        MethodInfo method = typeof(XmlSerializationReader).GetMethod(methodName, BindingFlags.Static | BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic, new Type[] { typeof(string) });
+                        return method.Invoke(this, new object[] { readFunc() });
+                }
+            }
+        }
+
+        object WriteStructMethod(StructMapping mapping, bool isNullable, bool checkType, string defaultNamespace)
+        {
+            if (mapping.IsSoap)
+                throw new PlatformNotSupportedException();
+            else
+                return WriteLiteralStructMethod(mapping, isNullable, checkType, defaultNamespace);
+        }
+
+        object WriteNullableMethod(NullableMapping nullableMapping, bool checkType, string defaultNamespace)
+        {
+            object o = Activator.CreateInstance(nullableMapping.TypeDesc.Type);
+            if (!ReadNull())
+            { 
+                ElementAccessor element = new ElementAccessor();
+                element.Mapping = nullableMapping.BaseMapping;
+                element.Any = false;
+                element.IsNullable = nullableMapping.BaseMapping.TypeDesc.IsNullable;
+
+                WriteElement(ref o, null, element, null, false, false, false, defaultNamespace);
+            }
+            else
+            {
+                
```

Comment:
Is there something that should be in this else? If there isn't, remove the else. Otherwise put something here or open an issue.